### ISSUE TITLE
Update to the latest version of test, test_api, and test_core

### ISSUE THREAD

**Issue by jakemac53** [flutter/flutter#81983]
The desired versions would be test_api version 0.4.0, test_core version 0.3.23, and test version 1.17.3.

This appears to currently be blocked on the usage of the pre-null-safety `mockito` package, which also requires an older version of `test_api`.

This prevents flutter users from getting any bug fixes or improvements from the `test` package (for instance a recent update to use frontend server for test compilation which dramatically improves vm test loading performance).

cc @mit-mit

**Comment by iapicca** [comment]
duplicate of https://github.com/flutter/flutter/issues/81294
